ldap/servers/slapd/haproxy.c

```diff
@@ -1,392 +1,874 @@
-
-done:
-        slapi_ch_free_string(&str_saved);
-    } else {
-        rc = HAPROXY_NOT_A_HEADER;
-    }
-    return rc;
-}
-
-/**
- * Function to receive and parse HAProxy headers, supporting both v1 and v2 of the protocol.
- * 
- * @param fd: The file descriptor of the socket from which to read.
- * @param proxy_connection: A pointer to an integer to store the proxy connection status (0 or 1).
- * @param pr_netaddr_from: A pointer to a PRNetAddr structure to store the source address info.
- * @param pr_netaddr_dest: A pointer to a PRNetAddr structure to store the destination address info.
- * 
- * @return: Returns 0 on successful operation, -1 on error.
- */
-int haproxy_receive(int fd, int *proxy_connection, PRNetAddr *pr_netaddr_from, PRNetAddr *pr_netaddr_dest)
-{
-    /* Buffer to store the header received from the HAProxy server */
-    char hdr[HAPROXY_HEADER_MAX_LEN + 1] = {0};
-    ssize_t recv_result = 0;
-    size_t hdr_len;
-    int rc = HAPROXY_ERROR;
-
-    /* Attempt to receive the header from the HAProxy server */
-    recv_result = recv(fd, hdr, sizeof(hdr) - 1,  MSG_PEEK | MSG_DONTWAIT);
-    if (recv_result <= 0) {
-        slapi_log_err(SLAPI_LOG_CONNS, "haproxy_receive", "EOF or error on haproxy socket: %s\n", strerror(errno));
-        return rc;
-    } else {
-        hdr_len = recv_result;
-    }
-
-    /* Null-terminate the header string */
-    if (hdr_len < sizeof(hdr)) {
-        hdr[hdr_len] = 0;
-    } else {
-        slapi_log_err(SLAPI_LOG_CONNS, "haproxy_receive", "Recieved header is too long: %ld\n", hdr_len);
-        rc = HAPROXY_NOT_A_HEADER;
-        return rc;
-    }
-
-    rc = haproxy_parse_v1_hdr(hdr, &hdr_len, proxy_connection, pr_netaddr_from, pr_netaddr_dest);
-    if (rc == HAPROXY_NOT_A_HEADER) {
-        rc = haproxy_parse_v2_hdr(hdr, &hdr_len, proxy_connection, pr_netaddr_from, pr_netaddr_dest);
-    }
-
-    if (rc == HAPROXY_HEADER_PARSED) {
-        slapi_log_err(SLAPI_LOG_CONNS, "haproxy_receive", "HAProxy header parsed successfully\n");
-        /* Consume the data from the socket */
-        recv_result = recv(fd, hdr, hdr_len, MSG_DONTWAIT);
-
-        if (recv_result != hdr_len) {
-            slapi_log_err(SLAPI_LOG_ERR, "haproxy_receive", "Read error: %s: %s\n", hdr, strerror(errno));
-            return HAPROXY_ERROR;
-        }
-    }
-    return rc;
-}
+/** BEGIN COPYRIGHT BLOCK
+ * Copyright (C) 2025 Red Hat, Inc.
+ * All rights reserved.
+ *
+ * License: GPL (version 3 or any later version).
+ * See LICENSE for details.
+ * END COPYRIGHT BLOCK **/
+
+/* haproxy.c - process connection PROXY header if present */
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include "slap.h"
+
+/* Function to parse IPv4 addresses in version 2 */
+static int haproxy_parse_v2_addr_v4(uint32_t in_addr, unsigned in_port, PRNetAddr *pr_netaddr)
```

Comment:
Not a big deal, we usually keep the function name and params on a newline versus the entire declaration all in one line